cocotb/share/lib/vpi/VpiImpl.h

```diff
@@ -266,6 +268,8 @@ class VpiImpl : public GpiImplInterface {
     VpiReadwriteCbHdl m_read_write;
     VpiNextPhaseCbHdl m_next_phase;
     VpiReadOnlyCbHdl m_read_only;
+    std::string m_product;
+    std::string m_version;
```

Comment:
Can these not go in `GpiImplInterface` as `protected`?

Reply:
Originally the implementations were different types but I later normalized them to strings, that more appropriate now.